go/iot/arduino/protocol.txt

```diff
@@ -0,0 +1,58 @@
+Protocol
+
+For raspberry Pi, it would firstly register the views and exporters. Only after the initial setup that Pi could respond
+to the record request from the arduino.
+
+For Arduino, after it collects sensor data, it encapsulates the sensor data based on the required Json format and sends
+record request to the Pi. Afterwards it could not send another record request until it receives a positive response from
+the Pi. If it receives a negative response, it would retry sending the request. Temporarily it would block when waiting
+for the response. We might add exponential back-off mechanism in the future.
+
+Every request or response should be ended with a newline character '\n'.
+
+The following example could illustrate the process.
+
+Raspberry Pi:
+
+When the program starts, the following view would be registered.
+
+	ProjectId : project-id
+	View:
+		Name : "opencensus.io/views/protocol_test"
+		Description : "View for Protocol Test"
+		Aggregation :
+        		AggregationType : view.LastValue()
+        		AggregationValue : []
+        Measure:
+        		Name : "my.org/measure/Measure_Test"
+        		Description : "Measure Test"
+        		Unit : 1
+        		MeasureType: int64
+        TagKeys: ["DeviceId", "SampleDate"]
+	ReportPeriod : 1
+
+Before the registration, the Raspberry Pi would make a negative response if there is any record request from Arduino.
+
+After the registration, the Raspberrry Pi would make a negative or positive response to any record request from Arduino.
```

Comment:
When would a negative response be sent? Same for positive response. It's nice to differentiate the two situations.